src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -199,20 +209,88 @@ def process(
                 if grant_statements is not None:
                     cc.message(grant_statements)
                     collected_output.append(grant_statements)
+            extension_fns_ddl = "\n".join(collected_output)
+            dest_file_py_file_to_ddl_map[py_file] = extension_fns_ddl
+
+        write_to_sql = kwargs.get(
+            "write_to_sql", True
+        )  # temporary until we have better e2e testing mechanism
+
+        if write_to_sql:
+            dest_file_to_sql_file = create_and_write_to_sql_files(
+                dest_file_py_file_to_ddl_map=dest_file_py_file_to_ddl_map,
+                deploy_root=self.deploy_root,
+                generated_root=self.generated_root,
+            )
 
-        return "\n".join(collected_output)
+            if len(dest_file_to_sql_file) > 0:
+                edit_setup_script_with_exec_imm_sql(
+                    dest_file_to_sql_file=dest_file_to_sql_file,
+                    deploy_root=self.deploy_root,
+                    generated_root=self.generated_root,
+                )
+
+        return dest_file_py_file_to_ddl_map
+
+    def _normalize_raw_imports(
+        self,
+        extension_fn: NativeAppExtensionFunction,
+        py_file: Path,
+        resolved_deploy_root: Path,
```

Comment:
the deploy root fed to the ctor is already resolved and absolute (IIRC, please check, but that was the intention)